### ISSUE TITLE
Problematic Accessibility behavior when upgrading to 0.70

### ISSUE THREAD

**Issue by snark** [snark/jumpcut#17]
Copying a report from issue #12:

"Removing and re-adding to Security/Accessibility was needed: only Jumpcut icon is visible in Accessibility panel, application name seems to be blank."

I've gotten multiple reports on this when upgrading. I'm not entirely sure what the cause is (my guess is that it has to do with 0.63 not being assigned formal Accessibility entitlements, but I'm not sure), and I am pretty sure that there's not anything to do to fix the issue with 0.70.

There's a one-time alert message that should appear for people upgrading from 0.63:

`Users upgrading from older versions of Jumpcut may need to remove Jumpcut from the Security & Privacy Preference's Accessibility tab and add it again manually in order to fix the "Clipping Selection Pastes" option. This should be a one-time fix.`

For any users upgrading from 0.63, is this message appearing? Could the instructions be clearer? Should this exact message be added to the website (and GitHub README) so people have a sense of what's going on, or should it be rephrased in some way?

**Comment by salbjerg** [comment]
Well, after *reading* your one-time alert:
I find that "Clipping Selection Pastes" is not easily understood.
(I did see that this is what it's now called in the Preferences).
So, putting your text through the old wringer, this version emerges:

In order to paste by clicking a copied item from Jumpcut's list, users
upgrading from older versions of Jumpcut will [may???] have to:

(1) go to System Preferences -> Security & Privacy -> Accessibility
(2) remove (the older version of) Jumpcut from the list
(3) add the new version of Jumpcut to the list
(4) click the tick-box to activate the permission.

ad 3) For hitherto unknown reasons, only Jumpcut's icon may show in the
list after the new version is added. Don't worry, it will work.

The statement "This should be a one-time fix." is probably superfluous.

I think adding alert text to READ ME and website will help users, most of
whom probably are upgraders at this time.
You could take it down later if you update the app, or after a year or two?

**Comment by AlfredENewmann** [comment]
Had trouble removing old Jumpcut from Accessibility.  Finally had to trash the old download.  Now new download (0.70) is operative (after granting it access)

**Comment by MaiYunfei2000** [comment]
https://github.com/snark/jumpcut/issues/12#issuecomment-527174177

> @MaiYunfei2000 You've checked the Accessibility tab, Jumpcut is present, and the checkbox is checked? If so, you may have hit some sort of weird bug caused by upgrading (you aren't the first one to report this), as I can confirm that the paste behavior working on 10.14.6. My suggestion is to open the Accessibility tab of the Security & Privacy pane, remove Jumpcut, and re-add it manually (click the "+" button, find the application).

> If that works for you, I'll add a note about this on the website.

Yes. I have checked and done this several times... I have also tried deleting the application and reinstall, but nothing work. 😅Strange... 

![image.png](https://i.loli.net/2019/09/03/yQTcdPAUxWqM8I1.png)

**Comment by itzawrap** [comment]
I just updated to .70 and checked to see if version .63 was in Accessibility. It wasn't. I guess I never put it in there. Added .70 and so far so good. Need to do a restart of iMac to check.

So glad you made this major upgrade because I was getting nervous about functionality under 10.15.x

Jumpcut makes my day so much more fun while clipping hundreds of items every week. Thanks for your continued support of this little gem

**Comment by snark** [comment]
@MaiYunfei2000 That icon without a name seems to be one of the signs people have reported of a bad Accessibility record. Try this (adapted from the Moom documentation):

1. Remove Jumpcut from the Accessibility list altogether. (Highlight it and click "-".)
2. Trash Jumpcut.
3. Restart your Mac.
4. Download a fresh copy of Jumpcut 0.70. (Here or using Homebrew.)
5. Install it.
6. See if the icon in Accessibility has the label "Jumpcut" next to it.

Thanks for your help -- I added your screenshot to a page on the site discussing this issue, as you are not the first to encounter it. (Just the one who seems to have gotten the worst case of it.)

**Comment by MaiYunfei2000** [comment]
Thank you!

But after I trying the following things, it still doesn't work.

1. Remove Jumpcut from the Accessibility list altogether. (Highlight it and click "-".)
2. Trash Jumpcut.
3. Restart my Mac.
4. Install Jumpcut via Homebrew but fail (and something strange happens, as the first image below shows, "jumpcut is already installed"; but I can't find it neither in the folder "Applications" nor the Accessibility tab).
5. Uninstall Jumpcut (see the first image).
6. Restart my Mac.
7. Install Jumpcut via Homebrew.
8. Add Jumpcut to the Accessibility list and click the tick-box to activate the permission.
9. Click the Jumpcut icon.
10. Click the "open" button of the alert window (the second image) and then nothing happens.

![Screen Shot 2019-09-04 at 16.20.13.png](https://i.loli.net/2019/09/04/EMyWHXzD68eLRvd.png)
![Screen Shot 2019-09-04 at 16.39.56.png](https://i.loli.net/2019/09/04/LaosUdmM3GZtgjq.png)

Something weird happens to my Mac. And I'm very sorry for bothering you so many times...

**Comment by snark** [comment]
Okay, focusing on this: "Install Jumpcut via Homebrew but fail (and something strange happens, as the first image below shows, "jumpcut is already installed"; but I can't find it neither in the folder "Applications" nor the Accessibility tab)." -- is it possible there's a copy of 0.63 in either your /Applications directory, ~/Applications, or... somewhere else on your hard drive?

And when you say nothing happens, you mean *nothing* happens--you don't get the scissors in the menu or anything? If you open up Console.app and filter for "Jumpcut", do you see any messages?

**Comment by MaiYunfei2000** [comment]
> is it possible there's a copy of 0.63 in either your /Applications directory, ~/Applications, or... somewhere else on your hard drive?

I've checked, it exists nowhere.

> you mean nothing happens--you don't get the scissors in the menu or anything?

I mean I don't get the scissors in the menu. 

When I open up Console.app as you instructed, I found much information about Jumpcut(see the image). But I can't understand this information...

![Screen Shot 2019-09-05 at 14.29.04.png](https://i.loli.net/2019/09/05/zmPpts3OdUMTRLo.png)
![Screen Shot 2019-09-05 at 14.30.43.png](https://i.loli.net/2019/09/05/Ct1oHldg7p3bEv8.png)
![Screen Shot 2019-09-05 at 14.30.16.png](https://i.loli.net/2019/09/05/8QftNoTas4IMu9W.png)

**Comment by snark** [comment]
@MaiYunfei2000 Okay, let me dwell on this a little. From the first screenshot, it looks like there might still be a copy of Jumpcut in your Trash? Can you check and, if so, empty it? The third screenshot is a stack trace, and it seems to be something going wrong with Sparkle, but I'm not sure what to make of it.

**Comment by MaiYunfei2000** [comment]
@snark Sorry, I forgot to mention that I've emptied the Trash before installing.

**Comment by snark** [comment]
Closing this issue out. I was never able to fully resolve the issue, which I *think* revolves around bundle identifiers in Security & Privacy; I hope it's no longer an issue with the improved instructions on the website and the popup on launch in 0.80 if Accessibility access is lacking.
